core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -45,13 +70,38 @@ case class SatoshisPerByte(currencyUnit: CurrencyUnit) extends BitcoinFeeUnit {
     SatoshisPerKiloByte(currencyUnit.satoshis * Satoshis(1000))
   }
 
-  override def txSizeForCalc(tx: Transaction): Long = tx.byteSize
+  override def txSizeForCalc(tx: Transaction): Long =
+    SatoshisPerByte.txSizeForCalc(tx)
 
-  override def scaleFactor: Long = 1
+  override def scaleFactor: Long = SatoshisPerByte.scaleFactor
```

Comment:
Seeing as this is the common pattern, I think you should have the top-level trait have an abstract def `FeeUnitFactory` and default impls like this ^ defined at the top level